ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationPool.java

```diff
@@ -31,16 +35,29 @@
 
 public class OperationPool<T extends SszData> {
   private static final int OPERATION_POOL_SIZE = 1000;
+  private static final String OPERATION_POOL_SIZE_METRIC = "operation_pool_size_";
+  private static final String OPERATION_POOL_SIZE_VALIDATION_REASON = "operation_pool_validation_";
   private final Set<T> operations = LimitedSet.create(OPERATION_POOL_SIZE);
   private final Function<UInt64, SszListSchema<T, ?>> slotToSszListSchemaSupplier;
   private final OperationValidator<T> operationValidator;
   private final Subscribers<OperationAddedSubscriber<T>> subscribers = Subscribers.create(true);
+  private final LabelledMetric<Counter> validationReasonCounter;
 
   public OperationPool(
+      String metricType,
+      MetricsSystem metricsSystem,
       Function<UInt64, SszListSchema<T, ?>> slotToSszListSchemaSupplier,
       OperationValidator<T> operationValidator) {
     this.slotToSszListSchemaSupplier = slotToSszListSchemaSupplier;
     this.operationValidator = operationValidator;
+
+    metricsSystem.createIntegerGauge(
+        TekuMetricCategory.BEACON, OPERATION_POOL_SIZE_METRIC + metricType, "result", this::size);
+    validationReasonCounter =
+        metricsSystem.createLabelledCounter(
+            TekuMetricCategory.BEACON,
+            OPERATION_POOL_SIZE_VALIDATION_REASON + metricType,
+            "result");
```

Comment:
```suggestion
            OPERATION_POOL_SIZE_VALIDATION_REASON + metricType,
            "Total number of attempts to add an operation to the pool, broken down by validation result",
            "result");
```

Reply:
done, thanks for this suggestion,  my only concern is if we need to pass non-zero number of labels metrics contract should have enforced this somehow.

Reply:
It doesn't have to be non-zero, just that the number of label names has to match the number of label values provided.  Unfortunately there isn't a good way of enforcing that at compile time while still supporting an arbitrary number of labels (and we ultimately inherit that API from Prometheus' library).